lib/apiservers/engine/backends/image.go

```diff
@@ -216,21 +203,45 @@ func (i *Image) Images(filterArgs string, filter string, all bool) ([]*types.Ima
 		return nil, err
 	}
 
+	// utilize the filterArgs to incorporate the argument option
+	if filter != "" {
+		imageFilters.Add("reference", filter)
+	}
+
 	// validate filters for accuracy and support
-	if err := vicfilter.ValidateFilters(imageFilters, acceptedImageFilterTags, unSupportedImageFilters); err != nil {
+	filterContext, err := vicfilter.ValidateImageFilters(imageFilters, acceptedImageFilterTags, unSupportedImageFilters)
+	if err != nil {
 		return nil, err
 	}
 
+	// get all images
 	images := cache.ImageCache().GetImages()
 
 	result := make([]*types.Image, 0, len(images))
 
-	for _, image := range images {
-		result = append(result, convertV1ImageToDockerImage(image))
-	}
+imageLoop:
+	for i := range images {
 
-	// sort on creation time
-	sort.Sort(sort.Reverse(byCreated(result)))
+		// provide filter with current ImageID
+		filterContext.ID = images[i].ImageID
+
+		// provide image labels
+		if images[i].Config != nil {
+			filterContext.Labels = images[i].Config.Labels
```

Comment:
No - The [filterCommon func](https://github.com/cgtexmex/vic/blob/vogue/lib/apiservers/engine/backends/filter/filter.go#L63) provides the common code that enables label filtering.

That same logic will be used by `docker ps`, `docker images`, `docker volume ls` and `docker network ls` -- we just have to ensure we are presenting the labels correctly.